vowpalwabbit/core/src/reductions/details/automl_impl.h

```diff
@@ -105,7 +111,7 @@ struct config_oracle
       std::vector<exclusion_config>& configs, const std::string& interaction_type, const std::string& oracle_type,
       std::shared_ptr<VW::rand_state>& rand_state);
 
-  void do_work(std::vector<std::pair<aml_estimator, estimator_config>>& estimators, const uint64_t current_champ);
+  void do_work(std::vector<challenger_champ_pair>& estimators, const uint64_t current_champ);
```

Comment:
oracle should be agnostic to estimator impl, candidate for removal/refactor. i.e. just send champs info instead of everything

Reply:
https://github.com/VowpalWabbit/vowpal_wabbit/pull/4117